processor/geoipprocessor/geoip_processor.go

```diff
@@ -5,26 +5,116 @@ package geoipprocessor // import "github.com/open-telemetry/opentelemetry-collec
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"net"
 
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/geoipprocessor/internal/provider"
 )
 
-type geoIPProcessor struct{}
+var errIPNotFound = errors.New("no IP address found in the resource attributes")
+
+// newGeoIPProcessor creates a new instance of geoIPProcessor with the specified fields.
+type geoIPProcessor struct {
+	providers          []provider.GeoIPProvider
+	resourceAttributes []attribute.Key
+}
+
+func newGeoIPProcessor(resourceAttributes []attribute.Key) *geoIPProcessor {
+	return &geoIPProcessor{
+		resourceAttributes: resourceAttributes,
+	}
+}
+
+// ipFromResourceAttributes extracts an IP address from the given resource's attributes based on the specified fields.
+// It returns the first IP address if found, or an error if no valid IP address is found.
+func ipFromResourceAttributes(attributes []attribute.Key, resource pcommon.Resource) (net.IP, error) {
+	for _, attr := range attributes {
+		if ipField, found := resource.Attributes().Get(string(attr)); found {
+			ipAttribute := net.ParseIP(ipField.AsString())
+			if ipAttribute == nil {
+				return nil, fmt.Errorf("could not parse ip address %s", ipField.AsString())
```

Comment:
Given that the [source.address](https://github.com/open-telemetry/semantic-conventions/blob/v1.26.0/docs/general/attributes.md#source) attribute can include a DNS name and not only IP, I don't think it's erroneous behavior to not be able to parse the value to an IP. I propose we ignore this error for now. We might add an error handling config option to the processor in the future, similar to Transform processor's [error_mode](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/README.md#config).

Reply:
Good point! I agree on skipping the error for now, added a comment to add the "error_mode" option in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33319/commits/3a332d823e25428e1390676f29933710a708134c